main/modes/factoryTest/factoryTest.c

```diff
@@ -0,0 +1,804 @@
+    int16_t binMargin = (TFT_WIDTH - (binWidth * FIX_BINS)) / 2;
+
+    // Find the max value
+    for (uint16_t i = 0; i < FIX_BINS; i++)
+    {
+        if (test->end.fuzzed_bins[i] > test->maxValue)
+        {
+            test->maxValue = test->end.fuzzed_bins[i];
+        }
+    }
+
+    // Plot the bars
+    int32_t energy = 0;
+    for (uint16_t i = 0; i < FIX_BINS; i++)
+    {
+        energy += test->end.fuzzed_bins[i];
+        uint8_t height       = ((TFT_HEIGHT / 2) * test->end.fuzzed_bins[i]) / test->maxValue;
+        paletteColor_t color = test->bzrMicPassed ? c050 : c500; // paletteHsvToHex((i * 256) / FIX_BINS, 255, 255);
+        int16_t x0           = binMargin + (i * binWidth);
+        int16_t x1           = binMargin + ((i + 1) * binWidth);
+        // Big enough, fill an area
+        fillDisplayArea(x0, TFT_HEIGHT - height, x1, TFT_HEIGHT, color);
+    }
+
+    // Check for a pass
+    if (energy > 100000)
+    {
+        test->bzrMicPassed = true;
+    }
+
+    // Draw button states
+    int16_t centerLine = TFT_HEIGHT / 4;
+    int16_t btnX       = BTN_DIA;
+
+    // PB_LEFT
+    plotButtonState(btnX, centerLine, test->buttonStates[2]);
+    btnX += DPAD_SEP;
+    // PB_UP
+    plotButtonState(btnX, centerLine - DPAD_SEP, test->buttonStates[0]);
+    // PB_DOWN
+    plotButtonState(btnX, centerLine + DPAD_SEP, test->buttonStates[1]);
+    btnX += DPAD_SEP;
+    // PB_RIGHT
+    plotButtonState(btnX, centerLine, test->buttonStates[3]);
+    btnX += BTN_DIA + 4;
+
+    // PB_SELECT
+    plotButtonState(btnX, centerLine, test->buttonStates[7]);
+    btnX += BTN_DIA + 2;
+    // PB_START
+    plotButtonState(btnX, centerLine, test->buttonStates[6]);
+    btnX += BTN_DIA + 4;
+
+    // PB_B
+    plotButtonState(btnX, centerLine + AB_SEP, test->buttonStates[5]);
+    btnX += BTN_DIA + 2;
+    // PB_A
+    plotButtonState(btnX, centerLine - AB_SEP, test->buttonStates[4]);
+
+    // Set up drawing accel bars
+    int16_t barY = TOUCHBAR_Y_OFF;
+
+    paletteColor_t accelColor = c500;
+    if (test->accelPassed)
+    {
+        accelColor = c050;
+    }
+
+    // Plot X accel
+    int16_t barWidth = ((test->x + 256) * MAX_ACCEL_BAR_W) / 512;
+    if (barWidth >= 0)
+    {
+        fillDisplayArea(TFT_WIDTH - barWidth, barY, TFT_WIDTH, barY + ACCEL_BAR_HEIGHT, accelColor);
+    }
+    barY += (ACCEL_BAR_HEIGHT + ACCEL_BAR_SEP);
+
+    // Plot Y accel
+    barWidth = ((test->y + 256) * MAX_ACCEL_BAR_W) / 512;
+    if (barWidth >= 0)
+    {
+        fillDisplayArea(TFT_WIDTH - barWidth, barY, TFT_WIDTH, barY + ACCEL_BAR_HEIGHT, accelColor);
+    }
+    barY += (ACCEL_BAR_HEIGHT + ACCEL_BAR_SEP);
+
+    // Plot Z accel
+    barWidth = ((test->z + 256) * MAX_ACCEL_BAR_W) / 512;
+    if (barWidth >= 0)
+    {
+        fillDisplayArea(TFT_WIDTH - barWidth, barY, TFT_WIDTH, barY + ACCEL_BAR_HEIGHT, accelColor);
+    }
+    barY += (ACCEL_BAR_HEIGHT + ACCEL_BAR_SEP);
+
+    // Draw the 8-direction touchpad with center circle
```

Comment:
I haven't ran this (at work) but I have a preference for drawing the analog dot-in-a-circle. It gives a better sense that analog touch is working as expected.